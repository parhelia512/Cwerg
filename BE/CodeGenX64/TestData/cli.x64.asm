# cli

.fun main NORMAL [S32] = [S32 A64]

.bbl start
    poparg argc_signed:S32
    poparg argv:A64

    conv argc:U32 argc_signed
    pusharg argc
    bsr print_u_ln

    mov i:U32 0
    bra check
.bbl loop
    shl index:U32 i 3
    ld  arg:A64 argv index
    pusharg arg
    bsr print_s_ln
    add i i 1
.bbl check
    blt i argc loop
    pusharg 0:S32
    ret
