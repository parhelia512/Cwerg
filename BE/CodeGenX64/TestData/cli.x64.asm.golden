7
build/cli.x64.asm.exe
1
2
3
aa
bbb
ccc
